pymc/distributions/distribution.py

```diff
@@ -355,205 +362,58 @@ def dist(
+    repeated function evaluations. This is used by `aesaraf.compile_pymc`.
 
-        Returns
-        -------
-        var : TensorVariable
-        """
+    If a `SymbolicRandomVariable` is registered as a `MeasurableVariable`, it is assumed
+    to also have a logprob method dispatched on its type. If not registered, the inner
+    graph will be automatically inlined when obtaining a logprob graph, so that Aeppl
+    can try to dynamically infer the varibale's logprob.
+    """
 
-        if "testval" in kwargs:
-            kwargs.pop("testval")
-            warnings.warn(
-                "The `.dist(testval=...)` argument is deprecated and has no effect. "
-                "Initial values for sampling/optimization can be specified with `initval` in a modelcontext. "
-                "For using Aesara's test value features, you must assign the `.tag.test_value` yourself.",
-                FutureWarning,
-                stacklevel=2,
-            )
-        if "initval" in kwargs:
-            raise TypeError(
-                "Unexpected keyword argument `initval`. "
-                "This argument is not available for the `.dist()` API."
-            )
+    _print_name = ("Unknown", "\\operatorname{Unknown}")
+    ndim_supp = None
 
-        if "dims" in kwargs:
-            raise NotImplementedError("The use of a `.dist(dims=...)` API is not supported.")
-        if shape is not None and size is not None:
-            raise ValueError(
-                f"Passing both `shape` ({shape}) and `size` ({size}) is not supported!"
-            )
+    def __init__(self, *args, ndim_supp, **kwargs):
+        self.ndim_supp = ndim_supp
+        kwargs.setdefault("inline", True)
+        super().__init__(*args, **kwargs)
 
-        shape = convert_shape(shape)
-        size = convert_size(size)
+    def update(self, node: Node):
+        return {}
 
-        ndim_supp = cls.ndim_supp(*dist_params)
-        create_size = find_size(shape=shape, size=size, ndim_supp=ndim_supp)
-        rv_out = cls.rv_op(*dist_params, size=create_size, **kwargs)
-        # This is needed for resizing from dims in `__new__`
-        rv_out.tag.ndim_supp = ndim_supp
 
-        rv_out.logp = _make_nice_attr_error("rv.logp(x)", "pm.logp(rv, x)")
-        rv_out.logcdf = _make_nice_attr_error("rv.logcdf(x)", "pm.logcdf(rv, x)")
-        rv_out.random = _make_nice_attr_error("rv.random()", "pm.draw(rv)")
-        return rv_out
+@node_rewriter([SymbolicRandomVariable])
+def inline_SymbolicRandomVariable(fgraph, node):
+    """
+    This optimization expands the internal graph of a SymbolicRV when obtaining logp
+    from Aeppl, in the case it does not have a specialized logp associated with it.
+    """
+    op = node.op
+    # If `Op` is not a `MeasurableVariable`, Aeppl must derive its logp from the inner graph
+    if not isinstance(op, MeasurableVariable):
+        return clone_replace(op.inner_outputs, {u: v for u, v in zip(op.inner_inputs, node.inputs)})
```

Comment:
Could clone_replace be dangerous? It might be cloning the input nodes and end up breaking the conditional dependence between input variables and outputs of the cloned graph

Reply:
I am just copying what the Aesara rewrite does so I assume not! 

https://github.com/aesara-devs/aesara/blob/d7fb940264a392857781b075e1b8178c603001a2/aesara/compile/builders.py#L932